src/plugins/custom_integrations/server/custom_integration_registry.test.ts

```diff
@@ -113,5 +134,44 @@ describe('CustomIntegrationsRegistry', () => {
         },
       ]);
     });
+
+    test('should filter integrations that need to replace EPR packages', () => {
+      const registry = new CustomIntegrationRegistry(mockLogger, true);
+      registry.registerCustomIntegration({ ...integration, id: 'bar', eprOverlap: 'aws' });
+      expect(registry.getAppendCustomIntegrations()).toEqual([]);
+    });
+  });
+
+  describe('getReplacementCustomIntegrations', () => {
+    test('should only return integrations with corresponding epr package ', () => {
+      const registry = new CustomIntegrationRegistry(mockLogger, true);
+      registry.registerCustomIntegration(integration);
+      registry.registerCustomIntegration({ ...integration, id: 'bar', eprOverlap: 'aws' });
+      expect(registry.getReplacementCustomIntegrations()).toEqual([
+        {
+          categories: ['upload_file'],
+          description: 'test integration',
+          icons: [],
+          id: 'bar',
+          isBeta: false,
+          shipper: 'tests',
+          title: 'Foo',
+          type: 'ui_link',
+          uiInternalPath: '/path/to/foo',
+          eprOverlap: 'aws',
+        },
+      ]);
+    });
+
+    test('should filter registrations without valid categories', () => {
```

Comment:
We'll be handling the dynamic additional categories in a separate PR, correct?

Reply:
yes. We pulled this out as separate work-item. It's more of a modeling improvement, less critical for showing/not-showing a beat. See https://github.com/elastic/kibana/pull/112330